src/Microsoft.Health.Fhir.TemplateManagement.FunctionalTests/TemplateCollectionFunctionalTests.cs

```diff
@@ -76,28 +78,37 @@ public static IEnumerable<object[]> GetValidImageInfoWithTag()
             yield return new object[] { new List<int> { 767, 838 }, "templatetest", "multilayers" };
         }
 
-        public static IEnumerable<object[]> GetHl7v2DataAndTemplateImageReference()
+        public static IEnumerable<object[]> GetHl7v2DataAndEntryTemplate()
         {
-            yield return new object[] { @"..\..\..\..\..\data\SampleData\Hl7v2\ADT01-23.hl7", "ADT_A01" };
-            yield return new object[] { @"..\..\..\..\..\data\SampleData\Hl7v2\IZ_1_1.1_Admin_Child_Max_Message.hl7", "VXU_V04" };
-            yield return new object[] { @"..\..\..\..\..\data\SampleData\Hl7v2\LAB-ORU-1.hl7", "ORU_R01" };
-            yield return new object[] { @"..\..\..\..\..\data\SampleData\Hl7v2\MDHHS-OML-O21-1.hl7", "OML_O21" };
-        }
-
-        public static IEnumerable<object[]> GetHl7v2DataAndTemplateImageReferenceWithoutGivenTemplate()
-        {
-            yield return new object[] { @"..\..\..\..\..\data\SampleData\Hl7v2\ADT01-23.hl7", "ADT_A01" };
-            yield return new object[] { @"..\..\..\..\..\data\SampleData\Hl7v2\IZ_1_1.1_Admin_Child_Max_Message.hl7", "VXU_V04" };
-            yield return new object[] { @"..\..\..\..\..\data\SampleData\Hl7v2\LAB-ORU-1.hl7", "ORU_R01" };
-            yield return new object[] { @"..\..\..\..\..\data\SampleData\Hl7v2\MDHHS-OML-O21-1.hl7", "OML_O21" };
+            var data = new List<object[]>
+            {
+                new object[] { @"ADT01-23.hl7", @"ADT_A01" },
+                new object[] { @"IZ_1_1.1_Admin_Child_Max_Message.hl7", @"VXU_V04" },
+                new object[] { @"LAB-ORU-1.hl7", @"ORU_R01" },
+                new object[] { @"MDHHS-OML-O21-1.hl7", @"OML_O21" },
+            };
+            return data.Select(item => new object[]
+            {
+                Path.Join(_sampleDataDirectory, "Hl7v2", Convert.ToString(item[0])),
+                Convert.ToString(item[1]),
+            });
         }
 
         public static IEnumerable<object[]> GetHl7v2DataAndTemplateSources()
         {
-            yield return new object[] { @"..\..\..\..\..\data\SampleData\Hl7v2\ADT01-23.hl7", @"..\..\..\..\..\data\Templates\Hl7v2", "ADT_A01" };
-            yield return new object[] { @"..\..\..\..\..\data\SampleData\Hl7v2\IZ_1_1.1_Admin_Child_Max_Message.hl7", @"..\..\..\..\..\data\Templates\Hl7v2", "VXU_V04" };
-            yield return new object[] { @"..\..\..\..\..\data\SampleData\Hl7v2\LAB-ORU-1.hl7", @"..\..\..\..\..\data\Templates\Hl7v2", "ORU_R01" };
-            yield return new object[] { @"..\..\..\..\..\data\SampleData\Hl7v2\MDHHS-OML-O21-1.hl7", @"..\..\..\..\..\data\Templates\Hl7v2", "OML_O21" };
+            var data = new List<object[]>
```

Comment:
Maybe data does not have to be List<object[]>, just two string lists is fine here